src/entities/protocols/uniswap.ts

```diff
@@ -56,10 +58,7 @@ export class UniswapTrade implements Command {
     // If the input currency is the native currency, we need to wrap it with the router as the recipient
     if (this.trade.inputAmount.currency.isNative) {
       // TODO: optimize if only one v2 pool we can directly send this to the pool
-      planner.addCommand(CommandType.WRAP_ETH, [
-        ROUTER_AS_RECIPIENT,
-        this.trade.maximumAmountIn(this.options.slippageTolerance).quotient.toString(),
-      ])
+      planner.addCommand(CommandType.WRAP_ETH, [ROUTER_AS_RECIPIENT, CONTRACT_BALANCE])
```

Comment:
What if there's a UNiswap trade that takes WETH, then an NFT trade that takes ETH? Would this break that?